### DISCUSSION TITLE
[Feature] Release clients with broad format support (so no need to transcode)

### DISCUSSION THREAD

**Discussion by pyorot** [immich-app/immich#10541] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

This is kind of a meta-feature to consider long-term. I think transcoding mainstream formats in general is a bit cringe for several reasons:
- Storage usage
- Electricity usage
- Quality loss
- Metadata loss (e.g. date created on file system)\*
- Longevity

\* (currently, originals are always kept with intact metadata, but there could be a transcode policy one day that replaces them to alleviate storage usage)

To expand that last point, there will come a time when browsers drop support for h264, because it's not supported in webm, and will stop being updated and become "insecure". Web compatibility is not the same as backward compatibility, because the web is designed to coerce web designers into a set of modern, maintained formats.

I think the use-case of immich is rather more similar to that of vlc, mpc etc., in that we're supporting people's archives of original media dating back decades. As such, I'd like to advocate for the approach of making clients widely compatible rather than transcoding to "web" standards. Jellyfin is a good role model, in that they release [clients that include universal media players](https://jellyfin.org/posts/client-jmp/) based on ffmpeg (via mpv). Hopefully, the maintenance burden of all these formats is outsourced into the media player being forked.

Common use-cases are, of course,
- Decades of legacy codecs (h261/mpeg1/mpeg2/h263/mpeg4 video specs, amr/mp2 audio),
- Unimplemented open standards (webm in safari)
- Non-guaranteed commonly-accepted formats (mov in chrome),
- License-encumbered stuff (hevc not working on firefox despite being licensed via the os), etc.

### Platform

- [ ] Server
- [X] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
An important note on the downsides you list for the current approach is that the original file is _always_ kept, and transcoding is only done for playback compatibility - so things like metadata loss aren't really a concern.

**Reply by pyorot** [reply]
very fair, added a note to explain there's a design tradeoff between metadata loss and storage usage

**Reply by bo0tzz** [reply]
Replacing the original files isn't something we'll ever build into core Immich. Someone might make a compression plugin or such, but that's not really our concern imo.
